### PR TITLE
Kp/regressor

### PR DESCRIPTION
**Pull request by kprokofi**
Added new functionality
Export works

### CODE REVIEW

File: torchdet3d/__init__.py [hunk 1]
```diff
@@ -2,11 +2,11 @@
 import sys
 import os
 
-module_path = os.path.abspath(os.path.join(os.path.dirname(__file__), '../3rdparty/Objectron'))
```
Comment (sovrasov) [inline torchdet3d/__init__.py:9]: Why do you need this change?

Comment (kprokofi) [reply torchdet3d/__init__.py:9]: because it runs with scripts in cache copied by setup.py and `__file__` gives the wrong path to __init__.py and it hasn't worked like that no matter how hard I try. If change it to `__init__.py` it works as expected.

File: torchdet3d/builders/model_builder.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            ''' use trained classificator to predict object's class and
+                choose according head for this '''
+            features = self.extract_features(x)
+            pooled_features = self._glob_feature_vector(features, mode=pooling_mode)
+            if num_classes > 1:
+                targets = self.classifier(pooled_features)
+            else:
+                targets = torch.zeros((x.size(0)), dtype=torch.long)
+
+            kp = torch.cat([self.regressors[id](sample) for id, sample in zip(targets, pooled_features)], 0)
```
Comment (sovrasov) [inline torchdet3d/builders/model_builder.py]: Why sigmoid is not applied here?

File: demo/demo.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    """Wrapper class for face detector"""
+    def __init__(self, ie,  model_path, conf=.6, device='CPU', ext_path=''):
+        self.net = load_ie_model(ie, model_path, device, None, ext_path)
+        self.confidence = conf
+        self.expand_ratio = (1., 1.)
+
+    def get_detections(self, frame):
+        """Returns all detections on frame"""
+        _, _, h, w = self.net.get_input_shape()
+        out = self.net.forward(cv.resize(frame, (w, h)))
```
Comment (sovrasov) [inline demo/demo.py]: `cv.resize` will be applied inside `self.net.forward`, so we don't need it here

File: demo/demo.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    def __init__(self, ie,  model_path, device='CPU', ext_path=''):
+        self.net = load_ie_model(ie, model_path, device, None, ext_path)
+
+    def get_detections(self, frame, detections):
+        """Returns all detections on frame"""
+        _, _, h, w = self.net.get_input_shape()
+        outputs = []
+        for i, rect in enumerate(detections):
+            cropped_img = self.crop(frame, rect[0])
+            out = self.net.forward(cv.resize(cropped_img, (w, h)))
```
Comment (sovrasov) [inline demo/demo.py]: The same thing about resize

File: demo/demo.py [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+from openvino.inference_engine import IECore
+
+from demo_tools import load_ie_model
+from torchdet3d.utils import draw_kp
+
+
+OBJECTRON_CLASSES = ('bike', 'book', 'bottle', 'cereal_box', 'camera', 'chair', 'cup', 'laptop', 'shoe')
+
+class Detector:
+    """Wrapper class for face detector"""
```
Comment (sovrasov) [inline demo/demo.py]: object detector

File: demo/demo.py [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+    capture.release()
+    cv.destroyAllWindows()
+
+def main():
+    """Prepares data for the 3d object detection demo"""
+
+    parser = argparse.ArgumentParser(description='3d object detection live demo script')
+    parser.add_argument('--video', type=str, default=None, help='Input video')
+    parser.add_argument('--cam_id', type=int, default=-1, help='Input cam')
+    parser.add_argument('--config', type=str, default=None, required=False,
```
Comment (sovrasov) [inline demo/demo.py:144]: Why do you need config here?

File: scripts/export.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    parser.add_argument('--config', type=str, default=None, required=True,
+                        help='path to configuration file')
+    parser.add_argument('--model_onnx_path', type=str, default='./converted_models/model.onnx', required=False,
+                        help='path where to save the model in onnx format')
+    parser.add_argument('--model_torch_path', type=str, required=False,
+                        help='path where to get the model in .pth format.'
+                             'By default the model will be obtained from config, the lastest epoch')
+    parser.add_argument('--model_mo_path', type=str, default='./converted_models', required=False,
+                        help='path where to save the model in IR format')
+    parser.add_argument('--convert_mo', type=bool, default=True, required=False,
```
Comment (sovrasov) [inline scripts/export.py:53]: `action='store_true'` would be more convenient here

File: demo/demo_tools.py [hunk 8]
```diff
@@ -0,0 +1,139 @@
+import cv2 as cv
+
+
+class IEModel:
```
Comment (sovrasov) [inline demo/demo_tools.py]: Seems like a part of content is duplicated here

File: torchdet3d/builders/model_builder.py [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+            pooled_features = self._glob_feature_vector(features, mode=pooling_mode)
+            if num_classes > 1:
+                targets = self.classifier(pooled_features)
+            else:
+                targets = torch.zeros((x.size(0)), dtype=torch.float32)
+
+            predicted_output = self.regressors[0](pooled_features).view(1, x.size(0), num_points // 2, 2)
+            if len(self.regressors) > 1:
+                for id, reg in enumerate(self.regressors[1:]):
+                    predicted_output = torch.cat((predicted_output, reg(pooled_features).view(1, x.size(0), num_points // 2, 2)))
```
Comment (sovrasov) [inline torchdet3d/builders/model_builder.py]: This approach to concatenation could slow down inference in IE. I'd rather put the outputs of all regressors to a list and then called `torch.cat` on it.

File: torchdet3d/utils/utils.py [hunk 10]
```diff
@@ -163,36 +176,64 @@ def load_pretrained_weights(model, file_path='', pretrained_dict=None, resume=Fa
...
 
-def draw_kp(img, keypoints, name, normalized=True, RGB=True, num_keypoints=9):
+def draw_kp(img, keypoints, name=None, normalized=True, RGB=True, num_keypoints=9, label=None):
     '''
     img: numpy three dimensional array
     keypoints: array like with shape [9,2]
     name: path to save
     '''
     img_copy = img.copy()
+    print(img_copy.shape)
```
Comment (sovrasov) [inline torchdet3d/utils/utils.py]: Please remove debug output

**Review by sovrasov — APPROVED** [review]
